### PR TITLE
[WIP] Add mogwaicoin, please add.

### PR DESCRIPTION
**Pull request by darkfriend77**
I haven't added the C# native neoscrypt implementation, because it will need to check unsafe code for compilation, if this isn't a problem I can add the source, otherwise it is on our github (https://github.com/mogwaicoin/NeoScryptCSharp).

I hope this fits your pull request requirements, if not let me know what is missing.

### CODE REVIEW

**Comment by darkfriend77** [comment]
Please add or tell us what we have to change ... to have the PR merged.

File: NBitcoin.Altcoins/Mogwai.cs [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+				return new MogwaiBlockHeader();
+			}
+			public override Block CreateBlock()
+			{
+				return new MogwaiBlock(new MogwaiBlockHeader());
+			}
+		}
+
+#pragma warning disable CS0618 // Type or member is obsolete
+		public class MogwaiBlockHeader : BlockHeader
```
Comment (NicolasDorier) [inline NBitcoin.Altcoins/Mogwai.cs]: you do not need this. Mogwai block format is same as bitcoin

File: NBitcoin.Altcoins/Mogwai.cs [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+			}
+
+			protected override HashStreamBase CreateHashStream()
+			{
+				return BufferedHashStream.CreateFrom(CalculateHash);
+			}
+
+		}
+
+		public class MogwaiBlock : Block
```
Comment (NicolasDorier) [inline NBitcoin.Altcoins/Mogwai.cs]: you do not need this.

File: NBitcoin.Altcoins/Mogwai.cs [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+		public static Mogwai Instance { get; } = new Mogwai();
+
+		public override string CryptoCode => "MOG";
+
+		private Mogwai()
+		{
+
+		}
+
+		public class MogwaiConsensusFactory : ConsensusFactory
```
Comment (NicolasDorier) [inline NBitcoin.Altcoins/Mogwai.cs]: you do not need this.

**Comment by darkfriend77** [comment]
thx for the reply, I removed unused code. In last commit.

File: NBitcoin.Altcoins/Mogwai.cs (hunk 1 above)
Comment (darkfriend77) [reply NBitcoin.Altcoins/Mogwai.cs]: okay, thank you for the hint I removed it in latest commit. It was copy paste from dash ... :-)

File: NBitcoin.Altcoins/Mogwai.cs (hunk 3 above)
Comment (darkfriend77) [reply NBitcoin.Altcoins/Mogwai.cs]: Thx, for the hint, I removed it in the latest commit, If you could accept merge request now, we would be happy. :-)

**Comment by NicolasDorier** [comment]
You tried to run the Altcoins tests with your node ?

**Comment by darkfriend77** [comment]
yes, I had issues with CanSignTransactions, CanSyncWithoutPoW, CanSyncWithPoW, CorrectCoinMaturity ... so I figured out it was because we had already activated DGW (drkGravityWaves Difficulty Algo) in reg test, which avoided generating more then 24 blocks without a diff increase.

So I changed that in regtest, but I also thought I would change BIP34, which seems to be wha did broke it again ... I gona fix that

**Comment by darkfriend77** [comment]
I will do a clean pull request once I got all test working. Sry for the lost time.

**Pull request closed without merging**